internal/service/appstream/directory_config.go

```diff
@@ -0,0 +1,205 @@
+package appstream
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/flex"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+func ResourceDirectoryConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceDirectoryConfigCreate,
+		ReadWithoutTimeout:   resourceDirectoryConfigRead,
+		UpdateWithoutTimeout: resourceDirectoryConfigUpdate,
+		DeleteWithoutTimeout: resourceDirectoryConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"directory_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"organizational_unit_distinguished_names": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringLenBetween(0, 2000),
+				},
+				Set: schema.HashString,
+			},
+			"service_account_credentials": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
```

Comment:
It seems not because when I set the parameter as optional, and then don't set the credentials, I got an error like this 
```
        Error: error creating AppStream DirectoryConfig (): InvalidParameterValueException: 1 validation error detected: Value null at 'ServiceAccountCredentials' failed to satisfy constraint: Member must not be null

```